File: 工程组件/命令系统/case_utils.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
案件通用功能模块
"""

import os
import glob

def get_case_path(base_dir, cases_dir, case_name):
    """
    获取案件路径（支持多种目录命名格式）

    参数:
        base_dir: 法律工程系统根目录
        cases_dir: 案例目录
        case_name: 案件名称

    返回:
        str: 案件路径，如果找不到则返回None
    """
    # 尝试多种可能的路径格式
    possible_paths = [
        os.path.join(cases_dir, f"案件：{case_name}"),  # 带"案件："前缀
        os.path.join(cases_dir, case_name),            # 无前缀
        os.path.join(base_dir, "案例", case_name)      # 绝对路径
    ]

    for path in possible_paths:
        if os.path.isdir(path):
            return path

    return None

def list_cases(cases_dir):
    """
    列出所有案件

    参数:
        cases_dir: 案例目录
    """
    print("案件列表:")
    # 查找带"案件："前缀的目录
    case_dirs = glob.glob(os.path.join(cases_dir, "案件：*"))
    # 查找无前缀的目录（排除"案件："前缀的目录）
    direct_cases = [d for d in glob.glob(os.path.join(cases_dir, "*"))
                  if os.path.isdir(d) and not os.path.basename(d).startswith("案件：")]

    case_dirs.extend(direct_cases)

    if not case_dirs:
        print("  (暂无案件)")
        return

    for case_dir in case_dirs:
        case_name = os.path.basename(case_dir).replace("案件：", "")
        print(f"- {case_name}")

def select_case(base_dir, cases_dir, case_name):
    """
    选择案件

    参数:
        base_dir: 法律工程系统根目录
        cases_dir: 案例目录
        case_name: 案件名称

    返回:
        bool: 是否选择成功
    """
    if not case_name:
        print("错误: 缺少案件名称参数")
        return False

    case_dir = get_case_path(base_dir, cases_dir, case_name)

    if not case_dir:
        print(f"错误: 案件 \"{case_name}\" 不存在")
        return False

    print(f"已选择案件: \"{case_name}\"")
    print(f"案件路径: {case_dir}")
    print(f"请使用 cd \"{case_dir}\" 进入案件目录")
    return True

File: src/utils/case_utils.js
/**
 * 案件通用功能模块
 */

const fs = require("fs");
const path = require("path");
const glob = require("glob");

/**
 * 获取案件路径（支持多种目录命名格式）
 *
 * @param {string} baseDir - 法律工程系统根目录
 * @param {string} casesDir - 案例目录
 * @param {string} caseName - 案件名称
 * @returns {string|null} - 案件路径，如果找不到则返回null
 */
function getCasePath(baseDir, casesDir, caseName) {
  // 尝试多种可能的路径格式
  const possiblePaths = [
    path.join(casesDir, `案件：${caseName}`), // 带"案件："前缀
    path.join(casesDir, caseName), // 无前缀
    path.join(baseDir, "案例", caseName), // 绝对路径
  ];

  for (const p of possiblePaths) {
    if (fs.existsSync(p) && fs.statSync(p).isDirectory()) {
      return p;
    }
  }

  return null;
}

/**
 * 列出所有案件
 *
 * @param {string} casesDir - 案例目录
 */
function listCases(casesDir) {
  console.log("案件列表:");

  // 确保案例目录存在
  if (!fs.existsSync(casesDir)) {
    console.log("  (暂无案件)");
    return;
  }

  // 查找带"案件："前缀的目录
  const prefixedCases = glob
    .sync(path.join(casesDir, "案件：*"))
    .filter((p) => fs.statSync(p).isDirectory());

  // 查找无前缀的目录（排除"案件："前缀的目录）
  const directCases = glob
    .sync(path.join(casesDir, "*"))
    .filter(
      (p) =>
        fs.statSync(p).isDirectory() && !path.basename(p).startsWith("案件：")
    );

  const caseDirs = [...prefixedCases, ...directCases];

  if (caseDirs.length === 0) {
    console.log("  (暂无案件)");
    return;
  }

  for (const caseDir of caseDirs) {
    const caseName = path.basename(caseDir).replace("案件：", "");
    console.log(`- ${caseName}`);
  }
}

/**
 * 选择案件
 *
 * @param {string} baseDir - 法律工程系统根目录
 * @param {string} casesDir - 案例目录
 * @param {string} caseName - 案件名称
 * @returns {boolean} - 是否选择成功
 */
function selectCase(baseDir, casesDir, caseName) {
  if (!caseName) {
    console.log("错误: 缺少案件名称参数");
    return false;
  }

  const caseDir = getCasePath(baseDir, casesDir, caseName);

  if (!caseDir) {
    console.log(`错误: 案件 "${caseName}" 不存在`);
    return false;
  }

  console.log(`已选择案件: "${caseName}"`);
  console.log(`案件路径: ${caseDir}`);
  console.log(`请使用 cd "${caseDir}" 进入案件目录`);
  return true;
}

module.exports = {
  getCasePath,
  listCases,
  selectCase,
};
